tests/integration.rs

```diff
@@ -310,3 +341,117 @@ fn usage() -> TestResult {
+
+    // Copy the whole nested-root-workspace folder (and its content) into the new temp dir,
+    // and then `cargo build` and run the sweep tests inside that directory.
+    fs_extra::copy_items(
+        &[&nested_workspace_dir],
+        temp_workspace_dir.path(),
+        &options,
+    )?;
+
+    let old_size = get_size(temp_workspace_dir.path())?;
+
+    // Build bin-crate
+    run(cargo(
+        temp_workspace_dir
+            .path()
+            .join("nested-root-workspace/bin-crate"),
+    )
+    // If someone has built & run these tests with CARGO_TARGET_DIR,
+    // we need to override that.
+    .env_remove("CARGO_TARGET_DIR")
+    .arg("build"));
+
+    let intermediate_build_size = get_size(temp_workspace_dir.path())?;
+    assert!(intermediate_build_size > old_size);
+
+    // Build workspace crates
+    run(
+        cargo(temp_workspace_dir.path().join("nested-root-workspace"))
+            // If someone has built & run these tests with CARGO_TARGET_DIR,
+            // we need to override that.
+            .env_remove("CARGO_TARGET_DIR")
+            .arg("build"),
+    );
+
+    let final_build_size = get_size(temp_workspace_dir.path())?;
+    assert!(final_build_size > intermediate_build_size);
+
+    // Measure the size of the nested root workspace and the bin crate before cargo-sweep is invoked.
+    let pre_clean_size_nested_root_workspace =
+        get_size(temp_workspace_dir.path().join("nested-root-workspace"))?;
+    let pre_clean_size_bin_create = get_size(
+        temp_workspace_dir
+            .path()
+            .join("nested-root-workspace/bin-crate"),
+    )?;
+
+    // Run a dry-run of cargo-sweep ("clean") in the target directory (recursive)
+    let args = &["-r", "--time", "0", "--dry-run"];
+    let expected_cleaned = clean_and_parse(args, |cmd| {
+        // If someone has built & run these tests with CARGO_TARGET_DIR,
+        // we need to override that.
+        cmd.env_remove("CARGO_TARGET_DIR")
+            .current_dir(temp_workspace_dir.path())
+    })?;
+    assert!(expected_cleaned > 0);
+    let size_after_dry_run_clean = get_size(temp_workspace_dir.path())?;
+    // Make sure that nothing was actually cleaned
+    assert_eq!(final_build_size, size_after_dry_run_clean);
+
+    // Run a proper cargo-sweep ("clean") in the target directory (recursive)
+    let args = &["-r", "--time", "0"];
+    let actual_cleaned = clean_and_parse(args, |cmd| {
+        // If someone has built & run these tests with CARGO_TARGET_DIR,
+        // we need to override that.
+        cmd.env_remove("CARGO_TARGET_DIR")
+            .current_dir(temp_workspace_dir.path())
+    })?;
+    assert_sweeped_size(temp_workspace_dir.path(), actual_cleaned, final_build_size)?;
+    assert_eq!(actual_cleaned, expected_cleaned);
+
+    // Measure the size of the nested root workspace and the bin crate after cargo-sweep is invoked,
+    // and assert that both of their sizes have been reduced.
```

Comment:
It took me a second to realize why this assertion passed.
```suggestion
    // and assert that both of their sizes have been reduced.
    // This works because by default cargo generates the `target/` directory in a subdirectory of the source dir.
```

Reply:
Counter-nit: `target/` doesn't end up in the "source dir", it ends up in the "package root" :)

Done